lib/engine.js

```diff
@@ -393,52 +393,51 @@ class Engine {
             };
 
          */
-        // if transformer.disableSourceDownload (defaults to false) then set it on input
-        const disableInputDownload = transformer.options && transformer.options.disableSourceDownload;
-        debug('disableInputDownload : ', disableInputDownload);
-        const generateUrl = (input && input.sourceType === TransformerSourceType.URL) 
-                || input.generateUrl; // input.generateUrl will go away
-
-        const downloadAsset = !disableInputDownload && !generateUrl;
-
         // create input object (type Asset)
         let transformerInput = {};
-        // asset needs to be downloaded locally
-        if(downloadAsset && input.url && !input.path) {
-
-            this[INTERNAL].timers.download.start();
-
-            // download input to <baseDirectory>/<stepNumber>-<transformerName>/inputfile and set input.path
-            transformerInput = await getSource(
-                input.url,
-                transformerDirectory.base,
-                false
-            );
-
-            this[INTERNAL].timers.download.stop();
-
-        } // url to asset is available and transformer to use it to process asset (ex: worker-ffmpeg)
-        else if(!downloadAsset && input.url) {
-            // dont download - T.compute will use url to process asset
-            // input.url is already set and is passed on to Asset object 
-            debug(`asset url available and disableInputDownload is ${disableInputDownload} : dont need to do anything`);
-            transformerInput = new Asset(input);
-        } // if (file exists locally && (T.manifest.input.sourceType === 'URL') then upload to tmp cloud
-        else if(!downloadAsset && input.path) {
-            const preSignedUrl = await this.getPresignedUrl(input.path);
-            // TODO: remove this log
-            debug('generated presigned url', preSignedUrl);
-            input.url = preSignedUrl;
-            transformerInput = new Asset(input, input.path);
-
-        } // asset is an intermediate rendition from previous transformer and is already available in input.path
-        else if(input.path) {
-            transformerInput = new Asset(input, input.path);
+        if (input && !input.url && !input.path) {
+            throw new GenericError('No source file accessible.');
         }
 
-        if (!transformerInput || !transformerInput.path) {
-            throw new GenericError('No source file accessible.');
+        debug(`sourceType: ${input.sourceType}`);
+        if (input.sourceType === TransformerSourceType.URL) {
+            const inputUrl =  input.url || '';
+            if (input.url && validUrl.isHttpsUri(inputUrl)) {
+                debug(`asset url available : dont need to do anything`);
+                transformerInput = new Asset(input);
+            } else if (input.url && isDataUrlAndValid(inputUrl)) {
```

Comment:
any reason, we check `isDataUrlAndValid` here but not at https://github.com/adobe/asset-compute-pipeline/pull/20/files#diff-6adce5113cc6b93b8f16a23dadb045ce21bb810dcb3e09bcf8578ba6ae16c608R436

Reply:
(I would also rename `isDataUrlAndValid` to `isValidDataUrl`).

Reply:
@pheenomenon I do have the check there, can you point out exactly which line?
I am basically following the code as is but with some refactoring.

Reply:
here: https://github.com/adobe/asset-compute-pipeline/pull/20/files#r719867088